File: frontend/lib/models/address.dart
class Address {
  Address(
      {required this.id,
      required this.name,
      required this.room,
      required this.floor,
      required this.building,
      required this.street,
      required this.district});

  final int id;
  final String name;
  final String room;
  final String floor;
  final String building;
  final String street;
  final String district;

  factory Address.fromJson(Map<String, dynamic> json) {
    return Address(
      id: json['id'],
      name: json['name'],
      room: json['room'],
      floor: json['floor'],
      building: json['building'],
      street: json['street'],
      district: json['district'],
    );
  }
}

File: backend/src/models/address.js
const { Sequelize, DataTypes } = require("sequelize");
const sequelize = require("../db/sequelize");

const Address = sequelize.define(
  "Address",
  {
    id: {
      type: DataTypes.INTEGER,
      allowNull: false,
      autoIncrement: true,
      primaryKey: true,
    },
    name: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    room: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    floor: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    building: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    street: {
      type: DataTypes.STRING,
      allowNull: false,
    },
    district: {
      type: DataTypes.STRING,
      allowNull: false,
    },
  },
  {
    freezeTableName: true,
  }
);

module.exports = Address;
